src/formatting-strings.md

```diff
@@ -0,0 +1,38 @@
+# Format Strings Using `%q`
+
+Whenever formatting messages that contain a string component via `fmt`, use `%q` instead of `%s`. This will wrap the specified string in quotes, helping it stand out from the rest of the error message. More importantly, if the string is empty, it will provide a more helpful error message.
```

Comment:
> Whenever formatting messages that contain a string component via `fmt`, use `%q` instead of `%s`.

I think we might want to clarify slightly: it's not necessarily just when a string component in general is used, but more when dynamic strings (incl. user-provided values) could potentially cause interpolation to result in something weird.

Maybe something like:
> Whenever dynamic strings - such as user-provided data - are used for formatting,
> prefer the placeholder `%q` over `%s`.